src/broker/forwarding_table.rs

```diff
@@ -60,17 +57,11 @@ impl ForwardingTable {
             for idx in 0..rx_count {
                 let (ref topic, ref body) = buf[idx];
                 
-                // retrieve the list of subscribers for this topic
-                let subscribers = match self.table.get(topic) {
-                    Some(subs) => subs,
-                    None => continue,
-                };
-            
                 // walk the list of clients that are subscribed to the topic
-                // and forward the message to them
-                for d in subscribers.iter() {
+                // prefix and forward the message to them
+                for d in self.topic_radix.find(topic) {
                     // skip if the publisher is trying to send to itself
-                    if Arc::<Client>::ptr_eq(p, d) {
+                    if Arc::<Client>::ptr_eq(p, &d) {
```

Comment:
We can drop this, since we are not simply broadcasting, if the subscriber end of a client is subscribed to a topic the publisher sends, we should still receive it